docs/plugin/interfaces/interfaces.md

```diff
@@ -76,6 +76,165 @@ By default, when the `@json` tag is added, all methods in the COM-RPC interface
 
 In older Thunder versions (<R4), JSON-RPC interfaces were defined using separate JSON schema files. These would then need to be manually wired up in the plugin. By using the code-generators, we can eliminate this step, making it much faster and easier to write plugins. It is no longer recommended to create JSON schema files for JSON-RPC interfaces.
 
+#### Overview
+
+* It is now possible to specify a separate JSON-RPC interface for handling connection issues and correct session management, and this will bring more options when the JSON-RPC interface deviates from the COM-RPC interface.
+
+* It is supported to send JSON-RPC over COM-RPC using the 'IDispatch' interface without the need for serialization.
+    * This is particularly useful for out-of-process (OOP) and remote plugins.
+
+* The JSON-RPC generator now supports the usage of nested "plain-old data"(POD) types (such as plain C structs) in the C++ interface.
+    * The JSON-RPC generator can now parse these structs and their usage in methods and generate a JSON-RPC interface for such cases.
+
+* `Core::hresult` is required as a return type value of all JSON-RPC enabled interfaces.
+
+* `@text` metatag has been extended to have more options to influence the names used in generated code.
+    * For more details click [here](../tags/#text).
+
+* Float type is now supported in the IDL header files.
+
+* `Core::instance_id` is now supported in the IDL header files.
+    * It is presented as a 32/64 bit hexadecimal value in JSON-RPC.
+
+* `Core::Time` is now supported in the IDL header files.
+    * It is presented as an [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) formatted string in JSON-RPC.
+
+* `Core::OptionalType<T>` allows a member to be optional (this superseded @optional), and must be used if an attribute is expected to be optional on JSON-RPC. In COM-RPC the OptionalType can be used to see if a value was set, and in JSON-RPC it is then allowed to omit this parameter.
+    * A @default tag can be used to provide a value, in the case T is not set. See more [here](../tags/#default).
+    * Note: Currently, OptionalType does not work with C-Style arrays.
+
+* JSON-RPC supports the usage of bitmask enums (a combination of enum values can be set in the same enum parameter at the same time).
+    * This is mapped as an array of values in the JSON-RPC interface.
+    * See more information about `/* bitmask */` [here](../tags/#bitmask).
```

Comment:
Shouldn't we refer to @encode:bitmask (and document thaT @bitmask is deprecated, guess we need to Ask sebastian)